tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -0,0 +1,518 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/client"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/resource"
+	"k8s.io/apimachinery/pkg/util/rand"
+	logger "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// log is the global logger for the client package. logf uses logr generic interface. Each log record produced
+// by this logger will have an identifier containing `client` tag.
+var log = logger.Log.WithName("aws-ec2")
+
+// awsProvider is a provider specific struct which contains clients for EC2, IAM, and the existing OpenShift
+// cluster that is running on EC2.
+// This is an implementation of the Cloud interface.
+type awsProvider struct {
+	// A client for EC2.
+	EC2 *ec2.EC2
+	// A client for IAM.
+	IAM *iam.IAM
+	// openShiftClient is the client of the existing OpenShift cluster.
+	openShiftClient *client.OpenShift
+	// resourceTrackerDir is where `windows-node-installer.json` file is stored.
+	resourceTrackerDir string
+}
+
+// New returns the AWS implementations of the Cloud interface.
+// credentialPath is a directory information to AWS credentials file.
```

Comment:
I think you mean `credentialPath is the directory where the AWS credentials file is stored`